vtest/response.go

```diff
@@ -0,0 +1,131 @@
+package vtest
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+)
+
+// Response is a simplified analog to http.Response. Its methods can be chained together to perform
+// multiple assertions on a single Response.
+type Response struct {
+	Body    []byte
+	Status  int
+	Headers http.Header
+	t       *testing.T
+}
+
+// AssertStatus asserts the HTTP status code of this Response
+func (r *Response) AssertStatus(status int) *Response {
+	r.t.Run("status", func(t *testing.T) {
+		if r.Status != status {
+			t.Errorf("got status %d, want %d", r.Status, status)
+		}
+	})
+
+	return r
+}
+
+// AssertHeader asserts the value of a single HTTP response header (case insenstive key)
+func (r *Response) AssertHeader(key, val string) *Response {
+	// normalize headers
+	h := http.Header{}
+	h.Add(key, val)
+
+	return r.AssertHeaders(h)
+}
+
+// AssertHeaders asserts the values of a map of HTTP response headers
+func (r *Response) AssertHeaders(headers http.Header) *Response {
+	for key := range headers {
+		r.t.Run(key, func(t *testing.T) {
+			val := headers.Get(key)
+			resv := r.Headers.Get(key)
+
+			if resv == "" {
+				r.t.Errorf("header %s: got <empty>, want '%s'", key, val)
+
+			} else if resv != val {
+				r.t.Errorf("header %s: got '%s', want '%s'", key, resv, val)
+			}
+		})
+
+	}
+
+	return r
+}
+
+// AssertBody asserts the response body is a byte-for-byte match
+func (r *Response) AssertBody(body []byte) *Response {
```

Comment:
I think I'd still like to see an `AssertJSON` that takes in `interface{}`, marshals that interface to JSON, and then compares bytes.

Reply:
Should we do some kind of normalization and uglification in case JSON keys are shuffled or there's extra whitespace?

Reply:
I'm fairly sure that Go's JSON marshaller is deterministic, but if you can find evidence to the contrary, let me know